libhoney.go

```diff
@@ -939,6 +939,10 @@ func (b *Builder) SendNow(data interface{}) error {
 // NewEvent creates a new Event prepopulated with fields, dynamic
 // field values, and configuration inherited from the builder.
 func (b *Builder) NewEvent() *Event {
+	return b.NewEventSized(0)
+}
+
+func (b *Builder) NewEventSized(size int) *Event {
```

Comment:
```suggestion
// NewEventSized() can be used instead of NewEvent() where the number of additional event fields is known and performance is a concern
// This will preallocate the internal field map based on the size and avoid unnecessary copying and reallocation of the map
func (b *Builder) NewEventSized(size int) *Event {
```